tests/bsim/bluetooth/samples/battery_service/src/central_test.c

```diff
@@ -58,6 +59,33 @@ static struct bt_gatt_subscribe_params battery_level_status_sub_params;
 
 static DEFINE_FLAG(notification_count_reached);
 static DEFINE_FLAG(indication_count_reached);
+static DEFINE_FLAG(bcs_char_read);
+
+/* Callback for handling Battery Critical Status Read Response */
+static uint8_t battery_critical_status_read_cb(struct bt_conn *conn, uint8_t err,
+					       struct bt_gatt_read_params *params, const void *data,
+					       uint16_t length)
+{
+	TEST_ASSERT(err == 0, "Failed to read Battery critical status (err %d)", err);
+
+	if (data) {
+		uint8_t status_byte = *(uint8_t *)data;
+
+		printk("[READ]  BAS Critical Status:\n");
+		printk("Battery state: %s\n", (status_byte & 0x01) ? "Critical" : "Normal");
+		printk("Immediate service: %s\n",
+		       (status_byte & 0x02) ? "Required" : "Not Required");
```

Comment:
Rather than using magic numbers `0x01` and `0x02` we could consider moving the definitions of `BCS_BATTERY_CRITICAL_STATE`and `BCS_IMMEDIATE_SERVICE_REQUIRED` to the public API so it would be easy for an application (like this) to read these values.